pyvrp/cpp/search/Route.h

```diff
@@ -114,6 +124,16 @@ class Route
      */
     [[nodiscard]] inline Duration timeWarp() const;
 
+    /**
+     * @return Time window segment corresponding to this route.
+     */
+    [[nodiscard]] inline TimeWindowSegment tws() const;
+
+    /**
+     * @return Total penalised cost of this route.
+     */
+    [[nodiscard]] inline Cost penalisedCost(CostEvaluator costEvaluator) const;
```

Comment:
I tried this but then `pyvrp::CostEvaluator` must depend on `pyvrp::search::Route` which I think is not what we want?

Reply:
It might be possible to template this. A simple `typename` should already work, but it's not bad if we can be more explicit. I could have a go at creating a type of `Evaluatable` [constraint](https://en.cppreference.com/w/cpp/language/constraints)? I haven't done this before but I kind of want to learn it, and I think this could be a good place to apply it.

Reply:
See #304 for an implementation of this. Should be plug-and-play with `pyvrp::search::Route` as well.

Reply:
Cool, I will try to update this after that's merged!